src/app/projects/new-project/new-project.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { ProjectsService } from '../projects.service';
+
+@Component({
+  selector: 'app-new-project',
+  templateUrl: './new-project.component.html',
+  styleUrls: [ './new-project.component.css' ]
+})
+export class NewProjectComponent implements OnInit {
+
+  id: number;
+  projectName: string;
+  description: string;
+  coefficient = 1.55;
+  error: string;
+
+  constructor(private router: Router,
+              private projectService: ProjectsService) {
+  }
+
+  ngOnInit() {
+  }
+
+  handleClick() {
+    if (this.projectName == null || this.description == null) {
```

Comment:
proponuję (!this.projectName || ...)
w js czasem zdarzy się jakiś undefined, a to jest coś innego (i łatwo przegapić takie błędy)